src/adapter/src/coord.rs

```diff
@@ -1558,9 +1559,28 @@ impl Coordinator {
                         .clone();
 
                     // Timestamp selection
-                    let as_of = self.bootstrap_materialized_view_as_of(&df_desc, mview.cluster_id);
+                    let as_of = self.bootstrap_materialized_view_as_of(
+                        &df_desc,
+                        mview.cluster_id,
+                        &mview.refresh_schedule,
+                    );
                     df_desc.set_as_of(as_of);
 
+                    // If we have a refresh schedule that has a last refresh, then set the `until` to the last refresh.
+                    let until = mview
+                        .refresh_schedule
+                        .as_ref()
+                        .map(|refresh_schedule| {
+                            refresh_schedule
+                                .last_refresh()
+                                .map(|last_refresh| last_refresh.try_step_forward())
+                        })
+                        .flatten()
+                        .flatten();
```

Comment:
I think this can be simplified:
```suggestion
                                        let until = mview
                        .refresh_schedule
                        .as_ref()
                        .and_then(|s| s.last_refresh())
                        .and_then(|r| r.try_step_forward());
```

Reply:
Ah, yes, thanks! Fixed in 1f4b5af3b8792d5bf05230d20dcd04a70dce827b